lib/index.js

```diff
@@ -1,11 +1,67 @@
+import path from 'path';
+import { readFile } from 'fs';
+import globby from 'globby';
 import postcss from 'postcss';
+import yaml from 'js-yaml';
 import Parser from './parser';
 
 export default postcss.plugin('postcss-map', opts => {
+  opts = opts || {};
+  opts.maps = opts.maps || [];
+  opts.basePath = opts.basePath || process.cwd();
+
+  let maps = Object.create(null);
+  let paths = opts.maps.filter(map => {
+    if (typeof map === 'string') {
+      return true;
+    }
+
+    if (typeof map === 'object') {
+      Object.assign(maps, map);
+    }
+  }).map(map => {
+    return path.resolve(opts.basePath, map);
+  });
+
+  let promise = globby(paths).then(paths => {
+    let filtered = [];
+    return paths.filter(map => {
+      let { name } = path.parse(map);
+      if (filtered.indexOf(name) === -1 && !maps[name]) {
+        filtered.push(name);
+        return true;
+      }
+    });
+  }).then(paths => {
+    return Promise.all(paths.map(map => {
+      return new Promise((resolve, reject) => {
+        readFile(map, 'utf-8', (err, data) => {
+          if (err) {
+            return reject(err);
+          }
+
+          try {
+            let { name } = path.parse(map);
+            maps[name] = yaml.safeLoad(data);
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        });
+      });
+    }))
+  });
+
   return css => {
-    const parser = new Parser(opts);
+    return promise.then(() => {
+      const parser = new Parser(maps);
 
-    css.walkDecls(::parser.processDecl);
-    css.walkAtRules(::parser.processAtRule);
+      css.walk(node => {
+        if (node.type === 'decl')
+          return parser.processDecl(node);
+        if (node.type === 'atrule')
+          return parser.processAtRule(node);
+      });
+    });
```

Comment:
Hm.. Don't understand you. I just reduced count of iterations. It's really good optimization. We do it in cssnano